app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt

```diff
@@ -342,6 +344,35 @@ enum class NavigationItem(
         override fun getRouteWithArgs(arguments: List<Any>): String = routeWithConversationIdArg(arguments)
     },
 
+    EditGuestAccess(
+        primaryRoute = EDIT_GUEST_ACCESS,
+        canonicalRoute = "$EDIT_GUEST_ACCESS?$EXTRA_CONVERSATION_ID={$EXTRA_CONVERSATION_ID}" +
+                "&$EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED={$EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED}" +
+                "&$EXTRA_EDIT_GUEST_ACCESS_IS_SERVICES_ALLOWED={$EXTRA_EDIT_GUEST_ACCESS_IS_SERVICES_ALLOWED}" +
+                "&$EXTRA_EDIT_GUEST_ACCESS_IS_UPDATING_GUEST_ACCESS_ALLOWED={$EXTRA_EDIT_GUEST_ACCESS_IS_UPDATING_GUEST_ACCESS_ALLOWED}",
+        content = { EditGuestAccessScreen() },
+        animationConfig = NavigationAnimationConfig.CustomAnimation(smoothSlideInFromRight(), smoothSlideOutFromLeft())
+    ) {
+        override fun getRouteWithArgs(arguments: List<Any>): String {
+            val conversationIdString: String = arguments.filterIsInstance<ConversationId>().firstOrNull()?.toString()
+                ?: "{$EXTRA_CONVERSATION_ID}"
+
+            val isGuestAccessAllowed: String = arguments.filterIsInstance<Boolean>().firstOrNull()?.toString()
+                ?: "{$EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED}"
+
+            val isServicesAllowed: String = arguments.filterIsInstance<Boolean>().firstOrNull()?.toString()
```

Comment:
Oh yes, it should  be `lastOrNull()` for service. That will work I would say

Reply:
Nope, because you actually pass three `Boolean` type params, so now it clashes with the next one - both `isServicesAllowed` and `isUpdatingGuestAccessAllowed` get the same `lastOrNull` value.
I still suggest that if we have to pass all these params then we should use a dedicated data class for that.

Reply:
yeah got your point, will use a data class for it then